src/compiler/factory.ts

```diff
@@ -3055,4 +3055,539 @@ namespace ts {
     function tryGetModuleNameFromDeclaration(declaration: ImportEqualsDeclaration | ImportDeclaration | ExportDeclaration, host: EmitHost, resolver: EmitResolver, compilerOptions: CompilerOptions) {
         return tryGetModuleNameFromFile(resolver.getExternalModuleFileFromDeclaration(declaration), host, compilerOptions);
     }
+
+    /**
+     * Transforms the body of a function-like node.
+     *
+     * @param node A function-like node.
+     */

```

Comment:
the only changes here are to add extra parameters needed to use transformFunctionBody and friends outside of a transformer, and to pass along convertObjectRest to `flattenParameterDestructuring`.